machine/v1/types_controlplanemachineset.go

```diff
@@ -287,6 +287,14 @@ type AzureFailureDomain struct {
 	// If nil, the virtual machine should be deployed to no zone.
 	// +kubebuilder:validation:Required
 	Zone string `json:"zone"`
+
+	// Subnet for the virtual machine.
+	// +optional
+	Subnet string `json:"subnet,omitempty"`
+
+	// InternalLoadBalancer is a name of the internal load balancer for the virtual machine.
```

Comment:
How come it's only the internal LB that matters here? 🤔

Reply:
The internal loadbalancer only accepts connections from the subnet it is in. There is now one NAT subnet per availability zone and one internal loadbalancer per zone. The public loadbalancer has an outbound facing IP address (Often internet routable), so there is only one.

Reply:
FYI, we're reverting back to a single internalLB and adding multiple frontends instead. From what I've seen, the cloud provider expects a load balancer named `$clusterName-internal` [1] so changing that causes problems. See [2]

[1] https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure_loadbalancer.go#L350
[2] https://github.com/openshift/installer/pull/7283/commits/30ef221c31f5d04a34abe61a9e128b6858dd4590

Reply:
Does that mean that we still need the subnet field, but not the internalLoadBalancer field?

Reply:
Frontend/backend pools we query